[Drop-In UI] add ability to observe when map click was not handled by drop-in

Zayankovsky opened the pull request:
### Description
Closes #6313. 

<!--
---------- CHECKLIST ----------
1. Add related labels (`bug`, `feature`, `new API(s)`, `SEMVER-MAJOR`, `needs-backporting`, etc.).
1. Adda a changelog entry under `Unreleased` tag or a `skip changelog` label if not applicable.
1. Update progress status on the project board.
1. Request a review from the team, if not a draft.
1. Add targeted milestone, when applicable.
1. Create ticket tracking addition of public documentation pages entry, when applicable.
-->

abhishek1508 commented on libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewListenerRegistry.kt:
```diff
@@ -48,6 +49,14 @@ internal class NavigationViewListenerRegistry(
 
     //endregion
 
+    fun onMapClicked(point: Point) {
+        for (listener in listeners.keys) {
+            if (listener.onMapClicked(point)) {
+                return
+            }
+        }
+    }
+
```
Lets us remove this function to keep the implementation consistent. Refs: https://github.com/mapbox/mapbox-navigation-android/pull/6360#discussion_r976669830

Zayankovsky commented:
> Let's add unit tests for `MapClickBehavior`

Added.
Also I updated the custom location puck used in `MapboxNavigationViewCustomizedActivity` to make it more fun, hope no one minds.
